Sources/ArrayExtensions.swift

```diff
@@ -201,6 +203,16 @@ extension Array where Element: Equatable {
     }
 }
 
+extension Array where Element: Hashable {
+
+    /// EZSE: Removes all occurrences of the given object(s)
+    public mutating func removeAll(_ elements: [Element]) {
+        let elementsSet = Set(elements)
```

Comment:
The bag transformation is not visible to the user: we only make use of the bag transformation locally for the supplied array argument (the one that tells us which elements in `self` that are to be removed), to allow `O(1)` lookup in the `filter` predicate below (`contains` applied to a `Set` instance rather than an `Array` instance). Since the supplied `elements` array are just identifiers we needn't duplicate them for the `self` removal lookup, even if the callee sends us an `Array` that contains duplicate elements. The result of the call to `removeAll(_:)` should be the same, but `O(n)` for `Hashable` objects and `O(n^2)` for those that are not. But I might've misunderstood your comment, please correct me if I'm wrong!